onnxmltools/convert/common/_topology.py

```diff
@@ -484,10 +487,10 @@ def _infer_all_types(self):
 
         # Traverse the graph from roots to leaves
         for operator in self.topological_operator_iterator():
-            if operator.type in self.custom_conversions:
-                cconv = self.custom_conversions[operator.type]
-                if isinstance(cconv, tuple) and len(cconv) > 1:  # user customized shape calculator.
-                    cconv[1](operator)
+            if operator.type in self.custom_shape_calculators:
+                self.custom_shape_calculators[operator.type](operator)
+            elif operator.type in self.custom_conversion_functions:
```

Comment:
It'd be better to have a single logic in the whole code base. As long as we have both of converter and shape calculator internally, I wish we can do this for customized operators. It's very confusing for new devs (if luckily we have more). Also, it doesn't make sense for scikit-learn not to register a shape calculator, which is allowed here..